test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/InputTagHelperTest.cs

```diff
@@ -113,6 +113,89 @@ public static TheoryData MultiAttributeCheckBoxData
+            var forAttributeName = "asp-for";
+
+            var expected = Resources.FormatInputTagHelper_InvalidExpressionResult(
+                "<input>",
+                forAttributeName,
+                possibleBool.GetType().FullName,
+                typeof(bool).FullName,
+                "type",
+                "checkbox");
+
+            var attributes = new TagHelperAttributeList
+            {
+                { "class", "form-control" },
+            };
+
+            var context = new TagHelperContext(
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+            var output = new TagHelperOutput(
+                originalTagName,
+                attributes,
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(result: null))
+            {
+                TagMode = TagMode.SelfClosing,
+            };
+            output.Content.AppendHtml(originalContent);
+            var htmlGenerator = new TestableHtmlGenerator(new EmptyModelMetadataProvider());
+            var tagHelper = GetTagHelper(htmlGenerator, model: possibleBool, propertyName: nameof(Model.IsACar));
+
+            // Act and Assert
+            var ex = Assert.Throws<InvalidOperationException>(() => tagHelper.Process(context, output));
+            Assert.Equal(expected, ex.Message);
+        }
+
+        [Theory]
+        [InlineData("trUE")]
+        [InlineData("FAlse")]
+        public void CheckBoxHandlesParsableStringsAsBoolsCorrectly(
+            string possibleBool)
+        {
+            // Arrange
+            var originalContent = "original content";
+            var originalTagName = "input";
+
+            var attributes = new TagHelperAttributeList
+            {
+                { "class", "form-control" },
+            };
+
+            var context = new TagHelperContext(
+                allAttributes: new TagHelperAttributeList(
+                    Enumerable.Empty<TagHelperAttribute>()),
+                items: new Dictionary<object, object>(),
+                uniqueId: "test");
+            var output = new TagHelperOutput(
+                originalTagName,
+                attributes,
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(result: null))
+            {
+                TagMode = TagMode.SelfClosing,
+            };
+            output.Content.AppendHtml(originalContent);
+            var htmlGenerator = new TestableHtmlGenerator(new EmptyModelMetadataProvider());
+            var tagHelper = GetTagHelper(htmlGenerator, model: possibleBool, propertyName: nameof(Model.IsACar));
+
+            // Act
+            var ex = Record.Exception(() => tagHelper.Process(context, output));
+            
+            // Assert
+            Assert.Null(ex);
```

Comment:
We intentionally have very few "does not throw" tests and definitely don't use `Record.Exception()` to confirm the call records nothing. So, please set up an expectation similar to https://github.com/Elfocrash/Mvc/blob/6edb9c9fce2b5d78ed3d533df71abd18495f8d5b/test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/InputTagHelperTest.cs#L86-L88 and assert that's what comes out.